Source/WebCore/css/CSSGridLineValue.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "CSSGridLineValue.h"
+
+#include <wtf/Vector.h>
+#include <wtf/text/StringBuilder.h>
+#include <wtf/text/WTFString.h>
+
+namespace WebCore {
+
+String CSSGridLineValue::customCSSText() const
+{
+    Vector<String> parts;
+    StringBuilder builder;
+    if (m_spanValue)
+        parts.append(m_spanValue->cssText());
+    // Only return the numeric value if not 1, or if it provided without a span value.
+    // https://drafts.csswg.org/css-grid-2/#grid-placement-span-int
+    if (m_numericValue) {
+        if (m_numericValue->intValue() != 1 || !m_spanValue)
+            parts.append(m_numericValue->cssText());
+    }
+    if (m_gridLineName)
+        parts.append(m_gridLineName->cssText());
+
+    builder.append(
+        interleave(parts, [](const String& part) { return part; }, " "_s)
```

Comment:
Do we want `WTFString::makeStringByJoining()` ?

Reply:
Sure, we can do that instead of manually calling StringBuilder.